hal/src/electron/modem/mdm_hal.cpp

```diff
@@ -1275,6 +1286,18 @@ int MDMParser::_cbURAT(int type, const char *buf, int len, bool *matched_default
 bool MDMParser::registerNet(const char* apn, NetStatus* status, system_tick_t timeout_ms)
 {
     LOCK();
+
+    // Set to full functionality mode
+    int cfun_val;
```

Comment:
Should be initialized to some default value, as `_cbCFUN` _might_ not modify it and leave it at an unknown value.

Reply:
Very true. thank u